feat(docs): passkey registration and authentication (#46340)
supabase/supabase: apps/docs/content/guides/auth/passkeys.mdx

</Admonition>

## How does it work?

Each sign-in or registration is a WebAuthn ceremony with three steps:

1. **Options**: the client requests a challenge from Supabase Auth.
2. **Ceremony**: the browser invokes `navigator.credentials.create()` (registration) or `navigator.credentials.get()` (authentication), prompting the user for biometrics or a security key.
3. **Verify**: the signed response is sent back to Supabase Auth, which validates the challenge and either stores the new credential or issues a session.

Supabase Auth uses [discoverable credentials](https://www.w3.org/TR/webauthn-3/#discoverable-credential) for sign-in. The user does not need to provide an email, phone, or username — the authenticator resolves the account from the credential it stores.

Registering a passkey requires an existing, confirmed, non-anonymous user. Sign-in works for any user that has previously registered a passkey, provided their email or phone is confirmed and the account is not banned.

## Enable passkey authentication

### Dashboard

> Open the [Passkeys settings](/dashboard/project/_/auth/passkeys) under **Authentication → Passkeys**, turn on **Enable Passkey authentication**, and fill in the WebAuthn [relying party](https://www.w3.org/TR/webauthn-3/#relying-party) details:

Reviewer: ```suggestion
Open the [Passkeys settings](/dashboard/project/_/auth/passkeys) from the  **Authentication → Passkeys** section of the Dashboard, turn on **Enable Passkey authentication**, and fill in the WebAuthn [relying party](https://www.w3.org/TR/webauthn-3/#relying-party) details:
```

A slow and unwritten attempt to standardise how we refere the Dashboard UI

> - **Relying Party Display Name**: a human-readable name for your application shown during the passkey prompt (for example `My App`).

Reviewer: ```suggestion
- **Relying Party Display Name**: a human-readable name for your application shown during the passkey prompt (for example, "My App").
```

> - **Relying Party ID**: the bare domain name for your application (for example `example.com`). Do not include a scheme, port, or path. This determines which passkeys can be used.

Reviewer: ```suggestion
- **Relying Party ID**: the bare domain name for your application (for example, "example.com"). Do not include a scheme, port, or path. This determines which passkeys can be used.
```

> - **Relying Party Origins**: comma-separated list of allowed origins (for example `https://example.com,https://app.example.com`). HTTPS is required except for loopback addresses (`localhost`, `127.0.0.1`, `[::1]`). Each origin's hostname must match or be a subdomain of the Relying Party ID. Up to 5 origins.

Reviewer: ```suggestion
- **Relying Party Origins**: comma-separated list of allowed origins (for example "https://example.com,https://app.example.com"). HTTPS is required except for loopback addresses ("localhost", "127.0.0.1", "[::1]"). Each origin's hostname must match or be a subdomain of the Relying Party ID. Up to 5 origins.
```

The dashboard pre-fills these from your project's Site URL and project name. Adjust them if your production app is served from a different domain.

<Admonition type="caution" title="Changing the Relying Party ID invalidates existing passkeys">

> Passkeys are cryptographically bound to the Relying Party ID they were registered against. Changing the RP ID makes every existing passkey unusable for sign-in, and users will need to register a new one. Pick the RP ID carefully before users start enrolling, and keep it stable once they do.

Reviewer: ```suggestion
Passkeys are cryptographically bound to the Relying Party (RP) ID they were registered against. Changing the RP ID makes every existing passkey unusable for sign-in, and users will need to register a new one. Pick the RP ID carefully before users start enrolling, and keep it stable once they do.
```

rp_display_name = "My App"
rp_id = "example.com"
rp_origins = ["https://example.com", "https://app.example.com"]
```

The `[auth.webauthn]` section is required when `auth.passkey.enabled` is `true`.

### Management API

You can also configure passkeys via the [Management API](/docs/reference/api/introduction):

```bash
# Get your access token from https://supabase.com/dashboard/account/tokens
export SUPABASE_ACCESS_TOKEN="your-access-token"
export PROJECT_REF="your-project-ref"

# Read the current passkey configuration
curl -X GET "https://api.supabase.com/v1/projects/$PROJECT_REF/config/auth" \
  -H "Authorization: Bearer $SUPABASE_ACCESS_TOKEN" \
  | jq '{passkey_enabled, webauthn_rp_id, webauthn_rp_display_name, webauthn_rp_origins}'

# Enable passkeys and set the WebAuthn relying party
curl -X PATCH "https://api.supabase.com/v1/projects/$PROJECT_REF/config/auth" \
  -H "Authorization: Bearer $SUPABASE_ACCESS_TOKEN" \
  -H "Content-Type: application/json" \
  -d '{
    "passkey_enabled": true,
    "webauthn_rp_display_name": "My App",
    "webauthn_rp_id": "example.com",
    "webauthn_rp_origins": "https://example.com,https://app.example.com"
  }'
```

## Enable in the client

> 

Reviewer: ```suggestion
<Admonition type="caution" title="Experimental">

```

> Passkey support is currently experimental and requires explicit opt-in as the API may change without notice:

Reviewer: ```suggestion
Passkey support is currently experimental and requires explicit opt-in as the API may change without notice.

</Admonition>
```

```ts
import { createClient } from '@supabase/supabase-js'

const supabase = createClient(supabaseUrl, supabaseKey, {
  auth: {
    experimental: { passkey: true },
  },
})
```

## Register a passkey

A user must be signed in before they can register a passkey. Typically, you call this from a security settings page, or directly after sign-up.

> `auth.registerPasskey()` runs the full WebAuthn ceremony: fetches a challenge, invokes the browser API, and verifies the response with Supabase Auth.

Reviewer: ```suggestion
`auth.registerPasskey()` runs the full WebAuthn ceremony. It fetches a challenge, invokes the browser API, and verifies the response with Supabase Auth.
```

```ts
const { data, error } = await supabase.auth.registerPasskey()

if (error) {
  // User cancelled, browser doesn't support WebAuthn, or verification failed
  console.error(error)
} else {
  console.log('Registered passkey', data.id)
}
```

The returned `data` contains the new passkey's metadata:

```ts
{
  id: string            // UUID — use this to update or delete the passkey
  friendly_name?: string // Derived from the authenticator's AAGUID
  created_at: string
}
```

> A friendly name is automatically derived from the authenticator's AAGUID (for example `iCloud Keychain`, `Google Password Manager`, `1Password`). Users can rename their passkey afterwards — see [Manage passkeys](#manage-passkeys).

Reviewer: ```suggestion
A friendly name is automatically derived from the authenticator's Authenticator Attestation GUID (AAGUID). For example, `iCloud Keychain`, `Google Password Manager`, `1Password`. Users can rename their passkey afterwards — see [Manage passkeys](#manage-passkeys).
```
